### PR TITLE
Celery

### PR DESCRIPTION
**Pull request by adambudziak**
Added celery and cleaned up some stuff.

### CODE REVIEW

File: .envs/.local/.backend [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CELERY_BROKER_URL=redis://redis:6379
+
+# Django cache
+REDIS_URL=redis://redis:6379
```
Comment (Mc01) [inline .envs/.local/.backend]: For future would be cool to leave trailing line at the end of the file.

Comment (Mc01) [reply .envs/.local/.backend]: Btw. What's the difference here between `CELERY_BROKER_URL` and `REDIS_URL`?

File: .envs/.local/.postgres [hunk 2]
```diff
@@ -0,0 +1,3 @@
+POSTGRES_DB=blogchain_db
+POSTGRES_USER=blogchain
+POSTGRES_PASSWORD=blogchain
```
Comment (Mc01) [inline .envs/.local/.postgres]: Trailing, same as above.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
-**/__pycache__
\ No newline at end of file
+**/__pycache__
+**/celerybeat.pid
+**/celerybeat-schedule
```
Comment (Mc01) [inline .gitignore]: Trailing 😂

File: assets/contracts.json [hunk 4]
```diff
@@ -1 +1 @@
-{"posts":"0x01A0bD87E68a3e2F1a6ef161679241756fa4DaBD","comment_store":"0xca36423723778BC799641738e2C17a0ad8DAd18C"}
\ No newline at end of file
+{"posts":"0x302cFcFC1e667a78f6f2E06AA8a36255a9BEF914","comment_store":"0x557f1271443824ECE86aC1a5409cFF00A2d15D55"}
```
Comment (Mc01) [inline assets/contracts.json]: You can add this to `.gitignore` I believe.

File: backend/posts/bc.py [hunk 5]
```diff
@@ -3,7 +3,6 @@
 import logging
```
Comment (Mc01) [inline backend/posts/bc.py]: File above (`backend/db.sqlite3`) should be not included in git ☝️

File: backend/posts/bc.py [hunk 6]
```diff
@@ -79,9 +77,10 @@ def get_comment(self, index):
...
     def verify_comment(self, comment: Comment, post: Post):
         comments_count = self.comments_count()
+        comment = CommentStoreContract.Comment(comment.data_hash, comment.post_hash)
         for comment_index in reversed(range(comments_count)):
             stored_comment = CommentStoreContract.Comment(*self.get_comment(comment_index))
-            if stored_comment.data_hash == comment.data_hash and stored_comment.post_hash == post.data_hash:
+            if stored_comment == comment:
                 comment.verified = True
                 comment.save()
                 break
```
Comment (Mc01) [inline backend/posts/bc.py]: In Python files you should not never leave no newline at the end (no trailing line).

File: backend/posts/serializers.py [hunk 7]
```diff
@@ -51,6 +51,8 @@ def validate(self, data):
         if computed_hash != data_hash:
             raise serializers.ValidationError('The data-hash is invalid.')
 
+        computed_hash = bytes.fromhex(computed_hash[2:])  # remove 0x
```
Comment (Mc01) [inline backend/posts/serializers.py]: Maybe just `computed_hash.lstrip('0x')`? Would be safer (applies if data is a string).

File: backend/start.sh [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -e
+
+python3 manage.py migrate
+python3 manage.py runserver 0.0.0.0:8000
```
Comment (Mc01) [inline backend/start.sh]: Trailing.

File: backend/start_celerybeat.sh [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+rm -f './celerybeat.pid'
+
+celery --app=taskapp beat --loglevel=INFO
```
Comment (Mc01) [inline backend/start_celerybeat.sh]: Trailing.

File: backend/start_celeryworker.sh [hunk 10]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+celery --app=taskapp worker --loglevel=INFO
```
Comment (Mc01) [inline backend/start_celeryworker.sh]: Trailing.

File: backend/start_flower.sh [hunk 11]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+celery flower \
+    --app=taskapp \
+    --broker="${CELERY_BROKER_URL}"
```
Comment (Mc01) [inline backend/start_flower.sh]: Trailing 😱

File: backend/taskapp/tasks.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    web3 = default_web3()
+    posts_address = get_contract_address('Posts')
+    posts_abi = get_contract_abi('Posts')
+    posts_contract = PostsContract(web3, posts_abi, posts_address)
+    verified_count = 0
+    for post in posts_to_verify:
+        if posts_contract.verify_post(post):
+            verified_count += 1
+
+    logging.info('Verified %d out of %d' % (verified_count, posts_to_verify.count()))
```
Comment (Mc01) [inline backend/taskapp/tasks.py]: https://unix.stackexchange.com/a/18789

Comment (Mc01) [reply backend/taskapp/tasks.py]: You need to memorise it now 😂 
Trailing

File: docker-compose.yml [hunk 13]
```diff
@@ -33,4 +63,7 @@ services:
       - backend
     volumes:
       - ./nginx:/etc/nginx
-      - ./assets:/var/www/blogchain/assets
\ No newline at end of file
+      - ./assets:/var/www/blogchain/assets
+
+volumes:
+  pgdata:
```
Comment (Mc01) [inline docker-compose.yml]: Trailing 🙀

File: frontend/src/components/CommentForm.js [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+          </Button>
+        </Form.Item>
+      </Form>
+    );
+  }
+}
+
+const WrappedCommentForm = Form.create({ name: 'create_comment' })(CreateCommentForm);
+
+export default connect(null, null)(WrappedCommentForm);
```
Comment (Mc01) [inline frontend/src/components/CommentForm.js]: The same as above :x

File: assets/contracts.json (hunk 4 above)
Comment (adambudziak) [reply assets/contracts.json]: It's already done in one of the PRs.

**Pull request merged**